contrib/babelfishpg_tsql/src/session.c

```diff
@@ -245,18 +245,21 @@ babelfish_db_name(PG_FUNCTION_ARGS)
 
 	if (dbid == 1)
 	{
-		dbname = palloc((strlen("master") + 1) * sizeof(char));
-		strncpy(dbname, "master", MAX_BBF_NAMEDATALEND);
+		int dbnamelen = strlen("master");
+		dbname = palloc0((dbnamelen + 1) * sizeof(char));
+		strncpy(dbname, "master", dbnamelen);
```

Comment:
Thanks for fixing this, we should also make the similar change in `set_cur_db` function above.

Reply:
Oh I checked this yesterday. `set_cur_db` is safe as the destination has proper length.

```
static char current_db_name[MAX_BBF_NAMEDATALEND + 1] = {'\0'};

...

void
set_cur_db(int16 id, const char *name)
{
    int         len = strlen(name);

    Assert(len <= MAX_BBF_NAMEDATALEND);

    current_db_id = id;
    strncpy(current_db_name, name, MAX_BBF_NAMEDATALEND);
    current_db_name[len] = '\0';

    if (*pltsql_protocol_plugin_ptr && (*pltsql_protocol_plugin_ptr)->set_db_stat_var)
        (*pltsql_protocol_plugin_ptr)->set_db_stat_var(id);
}
```